### ISSUE TITLE
Nullable reference type when using POCO classes raise CS8618 warning- can we avoid it?

### ISSUE THREAD

**Issue by moh-hassan** [dotnet/roslyn#40072]
I activated this feature in a project having POCO classes, as given below
```cs
 public class Connection
    {
        public string ServiceUrl { get; set; }
        public string? UserName { get; set; }
        public string? Password { get; set; }
      //... othere 
    }
```
and get the error:

> [CS8618] Non-nullable property 'ServiceUrl' is uninitialized. Consider declaring the property as nullable.

This is a POCO class, I don't initialize the class. This should be the responsibility of the one initializing the class to ensure that the properties are non-null. for example, the caller can do in other class/project:

var connection = new Connection
{
 ServiceUrl=some_value,
 //...
}

I had to avoid the warning by a workaround:

```cs
  public string ServiceUrl { get; set; }=default!
```
I expect Roslyn may do the same action by:
1)  using the forgiving operator! with get like this:

```cs
  public string ServiceUrl { get!; set; }  //append forgiving operator to get!.
//which can be translated to:
 public string ServiceUrl { get; set; }=default!
```
Or 2)  stop this warning for POCO classes (that with {get;set;}

**Comment by Joe4evr** [comment]
There's already a solution suggested at dotnet/csharplang#2328 (which is where you *should've* filed this).
